arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift

```diff
@@ -0,0 +1,171 @@
+        let minX = center.x - 0.5 * sideLength
+        let minY = center.y - 0.5 * sideLength
+        // The radius of the arcs.
+        let arcRadius = sideLength * 0.25
+        
+        // Bottom left curve.
+        let leftCurveStart = AGSPoint(x: center.x, y: minY, spatialReference: spatialReference)
+        let leftCurveEnd = AGSPoint(x: minX, y: minY + 0.75 * sideLength, spatialReference: spatialReference)
+        let leftControlPoint1 = AGSPoint(x: center.x, y: minY + 0.25 * sideLength, spatialReference: spatialReference)
+        let leftControlPoint2 = AGSPoint(x: minX, y: center.y, spatialReference: spatialReference)
+        let leftCurve = AGSCubicBezierSegment(start: leftCurveStart, controlPoint1: leftControlPoint1, controlPoint2: leftControlPoint2, end: leftCurveEnd, spatialReference: spatialReference)!
+        
+        // Top left arc.
+        let leftArcCenter = AGSPoint(x: minX + 0.25 * sideLength, y: minY + 0.75 * sideLength, spatialReference: spatialReference)
+        let leftArc = AGSEllipticArcSegment.createCircularEllipticArc(withCenter: leftArcCenter, radius: arcRadius, startAngle: .pi, centralAngle: -.pi, spatialReference: spatialReference)!
+        
+        // Top right arc.
+        let rightArcCenter = AGSPoint(x: minX + 0.75 * sideLength, y: minY + 0.75 * sideLength, spatialReference: spatialReference)
+        let rightArc = AGSEllipticArcSegment.createCircularEllipticArc(withCenter: rightArcCenter, radius: arcRadius, startAngle: .pi, centralAngle: -.pi, spatialReference: spatialReference)!
+        
+        // Bottom right curve.
+        let rightCurveStart = AGSPoint(x: minX + sideLength, y: minY + 0.75 * sideLength, spatialReference: spatialReference)
+        let rightCurveEnd = leftCurveStart
+        let rightControlPoint1 = AGSPoint(x: minX + sideLength, y: center.y, spatialReference: spatialReference)
+        let rightControlPoint2 = leftControlPoint1
+        let rightCurve = AGSCubicBezierSegment(start: rightCurveStart, controlPoint1: rightControlPoint1, controlPoint2: rightControlPoint2, end: rightCurveEnd, spatialReference: spatialReference)!
+        
+        let heart = [leftCurve, leftArc, rightArc, rightCurve].reduce(AGSMutablePart(spatialReference: spatialReference)) { (part: AGSMutablePart, segment: AGSSegment) -> AGSMutablePart in
+            part.add(segment)
+            return part
+        }
+        let heartShape = AGSPolygonBuilder(spatialReference: spatialReference)
+        heartShape.parts.add(heart)
+        return heartShape.toGeometry()
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["AddGraphicsWithRendererViewController"]
+        
+        // Add the graphics overlays, with graphics added, to the map view.
+        addGraphicsOverlays()
```

Comment:
If it were me, I would do away with `addGraphicsOverlay()` and instead have methods for creating each of the overlays and then call them in the `mapView` property observer:

```swift
mapView.graphicsOverlays.addObjects(from: [
    makePointGraphicsOverlay(),
    makeLineGraphicsOverlay(),
    makePolygonGraphicsOverlay(),
    makeCurveGraphicsOverlay()
])
```